pages/event/[id].js

```diff
@@ -0,0 +1,179 @@
+  }, [id]);
+
+  useEffect(() => {
+    if (role == USER.ROLES.GUARDIAN) {
+      getNinjas()
+        .then((response) => setNinjas(response.data))
+        .catch((error) => notification["error"](error.data?.errors));
+    }
+  }, []);
+
+  const registerNinjasOnEvent = () => {
+    selectedNinjas.map((ninja) =>
+      createEnrollment(ninja, id, false).catch((error) =>
+        notification["error"](error.data?.errors)
+      )
+    );
+  };
+
+  const registerMentorOnEvent = () => {
+    createAvailability(user.mentor_id, id, isAvailable, notes).catch((error) =>
+      notification["error"](error.data?.errors)
+    );
+  };
+
+  const eventTitle =
+    event?.title ||
+    `Sessão ${new Date(event?.start_time).toLocaleDateString("pt", {
+      weekday: "long",
+      year: "numeric",
+      month: "short",
+      day: "numeric",
+    })}`;
+
+  return (
+    <AppLayout>
+      <Row justify="start">
+        <Col {...{ xs: 24, md: 12, xl: 8, xxl: 8 }}>
+          <Title level={2}>{eventTitle}</Title>
+          <EventInfo
+            {...event}
+            breakpoints={{ xs: 1, sm: 1, md: 1, lg: 2, xl: 2, xxl: 2 }}
+          />
+          <Row justify="start" className="mt-4 mb-2">
+            {role == USER.ROLES.MENTOR ? (
+              <>
+                <Checkbox onChange={(e) => setAvailability(e.target.checked)}>
+                  Estás disponível?
+                </Checkbox>
+                <Input.TextArea
+                  rows={6}
+                  autoSize={{ minRows: 6, maxRows: 18 }}
+                  value={notes}
+                  onChange={(e) => setNotes(e.target.value)}
+                  placeholder="Alguma nota sobre a tua disponibilidade? Escreve-a aqui."
+                  className="mt-4 w-3/4"
+                />
+              </>
+            ) : (
+              <Select
+                mode="multiple"
+                className="w-3/4"
+                placeholder="Escolha pelo menos um Ninja"
+                onChange={setSelectedNinjas}
+                value={selectedNinjas}
+              >
+                {ninjas.map((ninja) => (
+                  <Select.Option key={ninja.id} value={ninja.id}>
+                    <div className="inline m-auto">
+                      <Avatar size={24} src={ninja.photo} />
+                      {`${ninja.first_name} ${ninja.last_name}`}
+                    </div>
+                  </Select.Option>
+                ))}
+              </Select>
+            )}
+          </Row>
+          {role == USER.ROLES.MENTOR ? (
+            <Button
+              type="primary"
+              className="mt-2 mb-4"
+              onClick={(_) => registerMentorOnEvent()}
+            >
+              Confirmar inscrição
+            </Button>
+          ) : (
+            <Button
+              type="primary"
+              className="mt-2 mb-4"
+              onClick={(_) => registerNinjasOnEvent()}
+            >
+              Confirmar inscrição
+            </Button>
```

Comment:
On click, this button should de disabled (on successful API call of course) or if the user is already registered, it should allow them to edit their availability if the enrollments are still open.

Also, on success the user should be redirected to the event index page

Look at #57 to see how I implemented a button that can only be clicked once every x seconds (that is called throttling)